File: app/src/main/java/com/example/tribecrypto/database/currency_details/DetailsDao.java
package com.example.tribecrypto.database.currency_details;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.tribecrypto.data.entity.CryptoCurrencyDetailsEntity;

import java.util.List;

import io.reactivex.Observable;

@Dao
public interface DetailsDao {

    @Query("SELECT * FROM crypto_details")
    Observable<List<CryptoCurrencyDetailsEntity>> getAllCryptoDetails();

    @Update
    void updateAll(List<CryptoCurrencyDetailsEntity> data);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAll(List<CryptoCurrencyDetailsEntity> data);

    @Query("DELETE FROM crypto_details")
    void deleteAllCryptoCurrenciesDetails();
}

File: app/src/main/java/com/example/tribecrypto/database/currency_details/DetailsDao.kt
package com.example.tribecrypto.database.currency_details

import androidx.room.*
import com.example.tribecrypto.data.entity.CryptoCurrencyDetailsEntity
import io.reactivex.Observable

@Dao
interface DetailsDao {

    @Query("SELECT * FROM crypto_details")
    fun getAllCryptoDetails(): Observable<List<CryptoCurrencyDetailsEntity>>

    @Update
    fun updateAll(data: List<CryptoCurrencyDetailsEntity>)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(data: List<CryptoCurrencyDetailsEntity>)

    @Query("DELETE FROM crypto_details")
    fun deleteAllCryptoCurrenciesDetails()
}